src/App.js

```diff
@@ -1,23 +1,61 @@
 import './App.css';
 import Form from "./components/Form/Form"
-import {useState} from "react"
+import {useEffect, useState} from "react"
 import {uid} from "uid"
 import List from "./components/List/List"
 import useLocalStorageState from "use-local-storage-state";
 
 function App() {
    const [activities, setActivities] = useLocalStorageState("activities", {defaultValue: []});
+   const [weather, setWeather] = useState()
+   const [temperature, setTemperature] = useState()
+   const [condition, setCondition] = useState()
 
    function handleAddActivity(newActivity) {
       setActivities([{...newActivity, id: uid()}, ...activities])
    }
+   
+   const goodWeatherActivities = activities.filter((activity) => activity.isForGoodWeather === true)
+   const badWeatherActivities = activities.filter((activity) => activity.isForGoodWeather === false)
 
+   useEffect(()=> {
+     async function fetchWeather() {
+      const response = await fetch("https://example-apis.vercel.app/api/weather");
+      const data = await response.json()
+      setWeather(data.isGoodWeather)
+      setTemperature(data.temperature)
+      setCondition(data.condition)
+    }
+    fetchWeather()
+   }, [])
+  
   return (
     <div className="App">
+      <>
+        {weather ? (
+          <>
+            <h1>
+              {condition} {temperature}°C
+            </h1>
+            <p>go outside the API is trolling you</p>
+            <List activities={goodWeatherActivities} />
+          </>
+        ) : (
+          <>
+            <h1>
+              {condition} {temperature}°C
+            </h1>
+            <p>stay inside and do something like:</p>
+            <List activities={badWeatherActivities} />
+          </>
+        )}
```

Comment:
This works, but you could write it even more concise if you had only one variable for the filtered activities.
After having that, this part of your code could look like this:

```js
          <>
            <h1>
              {condition} {temperature}°C
            </h1>
            <p>{weather ? "go outside the API is trolling you" : "stay inside and do something like:"}</p>
            <List activities={filteredActivities} />
          </>
```